### PR TITLE
Feature/iot 11 1/create alarm

### PR DESCRIPTION
**Pull request by THEWhiteArrow**
add:
- create alarm
- delete alarm
- native wrapper for delay function so it can be tested
- fix no perfect structure of external screen update time
- add scheduler to perform the tasks

### CODE REVIEW

File: lib/controllers/alarm.c [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+    while (shifted_minute >= 60)
+    {
+        shifted_hour++;
+        shifted_minute -= 60;
+    }
+    shifted_hour = shifted_hour % 24;
+
+    while (alarm_is_set && alarm_is_active && clock_hour == shifted_hour && clock_minute == shifted_minute)
+    {
+        // log_info("Alarm is now");
```
Comment (JakubCerovsky) [inline lib/controllers/alarm.c]: Cleanup.

File: lib/controllers/alarm.c [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+void alarm_stop()
+{
+    alarm_is_active = false;
+    alarm_is_stoped = true;
+    log_info("Alarm stopped");
+}
+
+void alarm_check()
+{
+    // log_info("Checking alarm");
```
Comment (JakubCerovsky) [inline lib/controllers/alarm.c]: Cleanup

File: lib/comm/ccp_message_handler.c [hunk 3]
```diff
@@ -9,9 +10,12 @@
 #include "leds.h"
 #include "ccp_message_sender.h"
 #include "message.h"
+#include "alarm.h"
 
 void ccp_handle_time_at(char *message);
 void ccp_handle_message_at(char *message);
+void ccp_handle_set_alarm(char *message);
```
Comment (JakubCerovsky) [inline lib/comm/ccp_message_handler.c]: Set alarm or create alarm?

**Review by JakubCerovsky — APPROVED** [review]
Great job, looks good just commented some minor things to maybe discuss or clean up. 👌🏼 all documentation is about creating alarm so naming it in code to set feels a bit unreasonable…

File: lib/drivers/external_screen.c [hunk 4]
```diff
@@ -123,4 +124,14 @@ void external_screen_static_with_scroll(const char *static_text, const char *scr
         }
     }
 }
+
+void external_screen_update_time_from_clock()
```
Comment (RubberDuck2307) [inline lib/drivers/external_screen.c]: can this method be please moved to display_controller. The drivers should be only operating the hardware, not handling logic such as getting the values from other files.

**Review by RubberDuck2307 — CHANGES_REQUESTED** [review]

File: src/main.c [hunk 5]
```diff
@@ -9,25 +9,38 @@
...
 	display_init();
-	external_screen_init();
-	clock_init();
+	scheduler_init();
+
+	int clock_minute_interval = 60;
+	// --- NOTICE ---
+	// Can be adjasted to mock time passing quicker. 60s = 1 minute, 1s = 1 second
+
+	scheduler_add_task(clock_update_time, clock_minute_interval);
```
Comment (RubberDuck2307) [inline src/main.c]: Can you please move adding the tasks to the `start() ` function of state_coordinator.

Comment (RubberDuck2307) [reply src/main.c]: I would like to keep separated the initialization that does not have a functionality from rest of the code that actually does something.

**Review by RubberDuck2307 — CHANGES_REQUESTED** [review]

**Review by RubberDuck2307 — APPROVED** [review]

**Pull request merged**